packages/website/docs/docs/block-types.md

```diff
@@ -102,4 +102,244 @@ type DefaultBlockProps = {
+
+As we saw in [Block Objects](/docs/blocks#block-objects), blocks can contain editable rich text which is represented as [Inline Content](/docs/inline-content). The `containsInlineContent` field allows your custom block to contain an editable rich-text field. For the custom image block, we use an inline content field to create our caption, so it's set to `true`.
+
+#### `render`
+
+This is a React component which defines how your custom block should be rendered in the editor, and takes two props:
+
+`block:` The block that should be rendered.
+
+`editor:` The BlockNote editor instance that the block is in.
+
+For our custom image block, we use a parent `div` which contains the image and caption. Since `block` will always be an `image` block, we also know it contains a `src` prop, and can pass it to the child `img` element.
+
+But what's this `InlineContent` component? Since we set `containsInlineContent` to `true`, it means we want to include an editable rich-text field somewhere in the image block. You should use the `InlineContent` component to represent this field in your `render` component. Since we're using it to create our caption, we add it below the `img` element.
+
+While the `InlineContent` component can be put anywhere inside the component you pass to `render`, you should make sure to only have one. If `containsInlineContent` is set to false, `render` shouldn't contain any.
+
+### Adding Custom Blocks to the Editor
+
+After creating `BlockConfig`s for all of our custom blocks, we need to convert them into `BlockSpec`s. A `BlockSpec` just tells BlockNote how your custom block should be defined internally, so we don't need to know much about it. We just need to convert our `BlockConfig` to a `BlockSpec` using the `createReactBlockSpec` function:
+
+```typescript jsx
+function createReactBlockSpec(blockConfig: BlockConfig): BlockSpec {...};
+```
+
+Now, all we need to do is pass it to the editor using the `blockSchema` option, which tells BlockNote which blocks to use. Let's again look at the image block from the demo as an example:
+
+```typescript jsx
+const ImageBlock = createReactBlockSpec(ImageBlockConfig);
+
+...
+
+// Creates a new editor instance.
+const editor: BlockNoteEditor<
+  DefaultBlockSchema & { image: typeof ImageBlock }
+> | null = useBlockNote({
+  // Tells BlockNote which blocks to use.
+  blockSchema: {
+    // Adds all default blocks.
+    ...defaultBlockSchema,
+    // Adds the custom image block.
+    image: ImageBlock,
+  }
+})
+```
+
+Notice three details about this code snippet. First, since we still want the editor to use the [Built-In Block Types](/docs/block-types#built-in-block-types), we add `defaultBlockSchema` to our custom block schema. Second, the key which we use for the custom image block is the same string we use for its type. Make sure that this is always the case for your own custom blocks. Finally, we provide the type of our schema as a type argument to `BlockNoteEditor`.
```

Comment:
is the last part necessary? I thought it would infer the type automatically

Reply:
For pretty much everything else the type is inferred automatically, but the type argument for `BlockNoteEditor` defaults to `DefaultBlockSchema`, so if no argument is provided, it uses that instead of inferring the type. Maybe that's something we should change?

Reply:
apparently typescript is smart enough for this. I'll remove the type annotation from the other branch